### PR TITLE
Scale clusters

### PR DESCRIPTION
**Pull request by brennancheung**
This feature still needs work but I am making it available for review now because the `<BlockChooser>` is needed for some other PR's that people are working on.

I will continue the work on this feature in another PR.

### CODE REVIEW

File: src/app/core/components/BlockChooser.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    padding: theme.spacing(1),
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  icon: {
+    maxWidth: '100px',
+  },
+  text: {
+    padding: `0 ${theme.spacing(2)}px`,
```
Comment (xantorres) [inline src/app/core/components/BlockChooser.tsx]: `padding: theme.spacing(0, 2)` is a bit more readable

File: src/app/plugins/kubernetes/components/infrastructure/clusters/ScaleMastersPage.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      }
+    </div>
+  )
+}
+
+const ScaleMastersPage = () => {
+  const classes = useStyles()
+  const { id } = useReactRouter().match.params
+  const [clusters, loading] = useDataLoader(clusterActions.list)
+  const cluster = (clusters || []).find(x => x.uuid === id)
```
Comment (xantorres) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/ScaleMastersPage.js]: you can get rid of the `|| []` data loaders should always return an array by default.

File: src/app/plugins/kubernetes/components/infrastructure/clusters/ScaleWorkersPage.tsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+          info="Number of worker nodes to deploy."
+          required
+          validations={[
+            validators.minValue(cluster.numWorkers),
+            validators.maxValue(cluster.numWorkers + MAX_SCALE_AT_A_TIME),
+          ]}
+        />
+      }
+
+      {!!cluster.enableCAS && (
```
Comment (xantorres) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/ScaleWorkersPage.tsx]: `!!` is unnecessary here

File: src/app/plugins/kubernetes/components/infrastructure/clusters/ScaleWorkersPage.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+const useStyles = makeStyles((theme: Theme) => ({
+  root: {
+    marginTop: theme.spacing(2),
+    display: 'flex',
+    flexFlow: 'row nowrap',
+    justifyContent: 'space-between',
+    maxWidth: '800px',
+  },
+  workerCount: {
+    margin: `${theme.spacing(4)}px 0px`,
```
Comment (xantorres) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/ScaleWorkersPage.tsx]: `theme.spacing(4, 0)`

File: src/app/core/components/BlockChooser.tsx (hunk 1 above)
Comment (xantorres) [inline src/app/core/components/BlockChooser.tsx]: `theme.spacing(0, 2)` is a bit more readable

File: src/app/plugins/kubernetes/components/infrastructure/clusters/ScaleWorkersPage.tsx (hunk 3 above)
Comment (brennancheung) [reply src/app/plugins/kubernetes/components/infrastructure/clusters/ScaleWorkersPage.tsx]: It's actually required because it was rendering the string `0` otherwise.

File: src/app/plugins/kubernetes/components/infrastructure/clusters/ScaleWorkersPage.tsx (hunk 4 above)
Comment (brennancheung) [reply src/app/plugins/kubernetes/components/infrastructure/clusters/ScaleWorkersPage.tsx]: Good to know.  Thanks.

File: .prettierrc.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+printWidth: 100
+tabWidth: 2
+useTabs: false
+semi: false
+singleQuote: true
+trailingComma: 'all'
+bracketSpacing: true
+jsxBracketSameLine: false
+arrowParens: 'always'
+proseWrap: 'preserve'
```
[the remark this answers is not in the document]
Comment (brennancheung) [reply .prettierrc.yaml]: Not on auto-save but for new files it saved me some effort.

File: src/app/plugins/kubernetes/components/infrastructure/clusters/ScaleWorkersPage.tsx (hunk 3 above)
Comment (jruddell) [reply src/app/plugins/kubernetes/components/infrastructure/clusters/ScaleWorkersPage.tsx]: we should always just use ternary operators for this logic. `0 && <span />` will render `0` into the browser.
